Use new localization tooling to generate base `.strings` for GlotPress and to download its translations

mokagio opened the pull request:
### Description

Update the Fastlane automation to generated the `.strings` file that gets uploaded to GlotPress and the related logic to download the localized `.strings` to use the new tooling.

It made sense to me to do both sides in one go because of the introduction of the `InfoPlist.strings` files which the `.strings` generation process appends to the base file, and the download process dispatches to the appropriate `.strings` for the app to read a runtime.

Continues the work from #6879. See also [WordPress iOS counterpart](https://github.com/wordpress-mobile/WordPress-iOS/pull/18061).

cc @oguzkocer always appreciate your review, but mostly wanted to flag this to keep an eye out on the `.strings` that will be generated during the next code freeze. I'll also try to check that PR when it's time for it.

### Testing instructions

Manually testing these changes is cumbersome. I don't expect reviewers to do it.

The steps I took to verify these changes were:

1. Create a test branch on top of this, https://github.com/woocommerce/woocommerce-ios/pull/6928, and run both automations on it
2. Create a test branch on top of `trunk` and run both legacy automation steps on it, https://github.com/woocommerce/woocommerce-ios/commit/ac0e8fce93a21181224eb4457856f69a6d2a7750 and https://github.com/woocommerce/woocommerce-ios/commit/188f913a5287f37d8129666bc8dd76e7dcd13a45
3. Ensure that all the changes in the generated `.strings` that the legacy automation produced are present in the new output
4. Compare the `de` `.strings` downloaded by both automations to make sure they are the same. _They weren't_, the new automation correctly removed a key the old one had missed.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

mokagio committed:
Use new `ios_generate_strings_file_from_code` workflow in `Fastfile`

This replaces the old and deprecated `ios_localize_project`

mokagio committed:
Add localized `InfoPlist.strings` files

They all contain the English placeholder strings at the moment. Our
tooling will soon rewrite them appropriately during the next code freeze
and release finalization cycle.

mokagio committed:
Rename `update_localization` lane to `generate_strings_file_for_glotpress`

This is clearer and in line with WordPress

mokagio commented on WooCommerce/Resources/ar.lproj/InfoPlist.strings:
```diff
@@ -0,0 +1,14 @@
+NSBluetoothPeripheralUsageDescription = "Bluetooth access is required in order to connect to supported card readers.";
+
+/* A message that tells the user why the app needs access to Bluetooth. */
+NSBluetoothAlwaysUsageDescription = "This app uses Bluetooth to connect to supported card readers.";
```
This are `Info.plist` keys that ought to be localized but hadn't been localized so far. 

You can see them in action by finding `MANUALLY_MAINTAINED_STRINGS_FILES` usages in the `Fastfile` below.

AliSoftware replied:
I'm guessing that for the purpose of this PR, you copied the `InfoPlist.strings` that you created for `en.lproj` to make a duplicate of it in all `*.lproj`, so that you'd have a starting point and an existing file on disk, so that you could all of those to the Xcode project? … even though in the next iteration, those files will be overwritten and regenerated in XML format by the new actions to contain the actual translations?

I'm fine with it, even though I wonder if it wouldn't have made sense to copy [the empty XML plist that the action will generated during next run](https://github.com/woocommerce/woocommerce-ios/pull/6928/files#diff-7c1e5780ebae6a5705e84873d6ac5fa806e3f2bd376569ae3bc9bf04176c369cR1-R6) instead, to make the very next diff on those files smaller and less surprising for Oguz and whoever review the future next-code-freeze PR… 🤷 

But that's not a big deal at all, since after the first iteration which will replace those with XML files, things will stabilize all by themselves anyway. 🙃

mokagio commented on fastlane/Fastfile:
```diff
@@ -146,7 +182,7 @@ platform :ios do
   desc 'Creates a new release branch from the current trunk'
   lane :complete_code_freeze do |options|
     ios_completecodefreeze_prechecks(options)
-    update_localization
+    generate_strings_file_for_glotpress
```
I took the chance to update this to what seems to me like a clearer name, which is the same as [what WordPress uses](https://github.com/wordpress-mobile/WordPress-iOS/blob/b8e95b1d57d6b3a3a4ab7501199b0216eadc3d03/fastlane/lanes/release.rb#L64)

mokagio commented on fastlane/Fastfile:
```diff
@@ -699,15 +735,26 @@ platform :ios do
 
   desc 'Downloads localized `.strings` from GlotPress'
   lane :download_localized_strings_from_glotpress do
-    # FIXME: This is a copy of what the release toolkit `ios_update_metadata` action does.
-    # We'll soon replace this with the new `ios_download_strings_files_from_glotpress`-based workflow.
```
_Feels good to address a `FIXME:` instead of always generating new ones._ 😅

mokagio committed:
Improve two `Fastfile` comments

AliSoftware commented on WooCommerce/WooCommerce.xcodeproj/project.pbxproj:
```diff
@@ -10121,6 +10141,30 @@
+			);
+			name = InfoPlist.strings;
+			sourceTree = "<group>";
+		};
```
I'm scaring myself a bit realizing that I was able to review and reason about that raw diff in this `project.pbxproj` in GitHub and make sense of it 😅 :
 - Adding the files to the Xcode project [added the `PBXFileReferences` for each individual `*.lproj/Info.plist` to the object list](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-e1be82bcbaa4c150399e0fe9151d71246d28888cc56af37a925ee7545c6fa5b7R2377-R2393), then [created a `PBXVariantGroup` grouping those localized variants and referencing them as children](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-e1be82bcbaa4c150399e0fe9151d71246d28888cc56af37a925ee7545c6fa5b7R10144-R10167), which itself was [added as a child of its parent folder/group in the Xcode folders hierarchy](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-e1be82bcbaa4c150399e0fe9151d71246d28888cc56af37a925ee7545c6fa5b7R6326)
 - Then adding that `*.strings` variant group to the "Compile Sources" build phase [added a `PBXBuildFile` to the objects list, referencing the `PBXVariantGroup`](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-e1be82bcbaa4c150399e0fe9151d71246d28888cc56af37a925ee7545c6fa5b7R659), then [added that `PBXBuildFile` to the list in the `PBXSourceBuildPhase`](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-e1be82bcbaa4c150399e0fe9151d71246d28888cc56af37a925ee7545c6fa5b7R8435) to make them be compiled as part of this build phase.

TL;DR: Help, I can see in the matrix 😂 
For sure my work on the `LintAppLocalizedStringsUsage.swift` linter in WPiOS helped me _a lot_ in there… but I think the side effect will be that I'll get better at understanding diffs on `pbxproj` files now (and better at fixing conflicts on them, I guess!)… always useful to know the inner workings of Xcode after all 🙂

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -29,9 +28,46 @@ SCREENSHOT_DEVICES = [
+# List of `.strings` files manually maintained by developers (as opposed to being automatically extracted from the code)
+# which we will merge into the main `Localizable.strings` file imported by GlotPress, then extract back once we download the translations.
+#
+# Each `.strings` file to be merged/extracted is associated with a prefix to add the the keys being used to avoid conflicts and differentiate.
```
```suggestion
# Each `.strings` file to be merged/extracted is associated with a prefix to add to the keys, used to avoid conflicts and differentiate the source of the copies.
```

(I'm guessing you copy/pasted those comments from WPiOS? If so, I'm guessing I made a similar typo in WPiOS too 🤷  So feel free to fix it there too)

mokagio replied:
> I'm guessing you copy/pasted those comments from WPiOS?

Yep 😅 I edited some bit here and there, but clearly missed that one. Good catch, thank you!

mokagio replied:
> So feel free to fix it [the typo] there too

Good point. Done: https://github.com/wordpress-mobile/WordPress-iOS/pull/18769

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -29,9 +28,46 @@ SCREENSHOT_DEVICES = [
+#
+# See calls to `ios_merge_strings_files` and `ios_extract_keys_from_strings_files` for usage.
+#
+# Note that, yes, we currently have only one file, but it's still worth defining this constant because it's used across more than one lane, as describe above.
```
👍 
Also it allows us to _document_ that constant in lots of details like you did here, and makes it very clear about the intent by using a clear name for it — win-win!

AliSoftware replied:
```suggestion
# Note that, yes, we currently have only one file, but it's still worth defining this constant because it's used across more than one lane, as described above.
```

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -699,15 +735,26 @@ platform :ios do
+    )
+    git_commit(
+      path: File.join(RESOURCES_FOLDER, '*.lproj', 'Localizable.strings'),
+      message: 'Update app translations – `Localizable.strings`',
```
❤️  Clear commit messages for the win!

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -785,7 +832,7 @@ platform :ios do
     }
 
-    metadata_locales = GLOTPRESS_LOCALES
+    metadata_locales = GLOTPRESS_TO_ASC_METADATA_LOCALE_CODES
```
⚠️ I think that thanks to that change (using the Hash which maps GP codes to ASC codes, instead of using just an array of GP codes) should make you able to delete the lines [L796–L813](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688L796-L813) below now

AliSoftware replied:
ℹ️  Note that @rachelmcr is working on fixing some stuff in the Screenshots, and might soon modify [those lines a bit below that change](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688R862-R877) as well, to fix a mismatch in the `promo_screenshot_*.txt` file names.

Or maybe not, depending on how your other PRs https://github.com/woocommerce/woocommerce-ios/pull/6939 and https://github.com/woocommerce/woocommerce-ios/pull/6937 evolve and if you take the occasion to fix those inconsistencies directly while addressing my feedback in those PRs of yours…

@mokagio I'll let you and @rachelmcr synchronize directly to make sure you are not stepping on each other's toes while making those similar/correlated changes in each of your PRs.

rachelmcr replied:
Thanks for looping me in to these PRs, @AliSoftware! I wasn't aware that these metadata changes were also in the works right now.

@mokagio If it's easier for you, I can avoid making any changes to the metadata file names myself, at least for now. I have other fixes for screenshots in https://github.com/woocommerce/woocommerce-ios/pull/6933, and I can stick to the fixes already done there. That way we can coordinate the `promo_screenshot_*.txt` file names separately — either making sure the issue I ran into is resolved with the changes you're working on or following up to fix that afterwards.

mokagio replied:
Thanks for coordinating @AliSoftware 🙇‍♂️ 

@rachelmcr 👋 Yeah, I think it'd be safer to leave file name changes for later once we got all the different tooling changes in. "That way we can coordinate the `promo_screenshot_*.txt` file names separately" 👍

mokagio replied:
I opened https://github.com/woocommerce/woocommerce-ios/issues/6987 to track the various steps.

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -911,8 +958,38 @@ platform :ios do
+
+    git_commit(
+      path: en_lproj_path,
+      message: 'Update strings for localization',
```
Super-nit: I wonder if we shouldn't start using a term like "freeze" / "string freeze" as a parallel to "code freeze" (here and ideally ultimately in the other repos too, of course, including Android which has slightly different implementation details for that step but does the correpsonding steps with the same intention as here).

After all, what we're doing here is "freezing" the strings that are in code, the same way we freeze the code itself (except that given GlotPress imports files that are in `trunk` and not in `release/*` branches, we need to only update those `.strings` files during "code (& strings) freeze").

That might be a bit more explicit in the Android side, because for Android what we do is that we just _copy_ the `src/main/res/values/strings.xml` file into `fastlane/resources/values/strings.xml` (which is the file that gets imported by GlotPress), so it might be easier/more natural to see that copy of the file as a "freeze" of said file. While here for iOS we don't "make a copy of a file to freeze it", but instead "only generate the strings file from code during code freeze". But still, the idea is the same, and `"Freeze strings for translation"` might thus be a more explicit commit message?

wdyt?

mokagio replied:
I like the symmetry 👍

mokagio replied:
```suggestion
      message: 'Freeze strings for localization',
```

AliSoftware reviewed (changes requested):
Nice one!

Mostly ok… except for the fact that I think this PR should also delete [the lines that tries to move files from GP-locale-code-based folders to ASC-locale-code-based folders](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688L796-L813) now that you migrated to use the `GLOTPRESS_TO_ASC_METADATA_LOCALE_CODES` mapping instead of just the `GLOTPRESS_LOCALES` array — hence me submitting this review as "Request changes".

mokagio committed:
Use "freeze" instead of "update" for clarity

See conversation at https://github.com/woocommerce/woocommerce-ios/pull/6936/files#r881839918

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio committed:
Fix typo in localization tooling documentation comment

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio committed:
Remove unnecessary files operation from `download_app_store_translations`

The changes in 066b80acb2bc398ea7c188a122d74eab21447b63 made it
unnecessary because we now use a locales dictionary that has all the
appropriate values.

mokagio committed:
Remove call that deleted `metadata/default`, now the source of truth

Made the source of truth in f2f92bd3bced44e4d8259ffb50668f9c1bd95d47

mokagio commented:
@AliSoftware 

> this PR should also delete [the lines that tries to move files from GP-locale-code-based folders to ASC-locale-code-based folders](https://github.com/woocommerce/woocommerce-ios/pull/6936/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688L796-L813) now that you migrated to use the GLOTPRESS_TO_ASC_METADATA_LOCALE_CODES mapping instead of just the GLOTPRESS_LOCALES array

Addressed in 3bd6390f4. Thank you for picking this up, that code was now failing because it tried to read from folders that weren't there because of the locales dictionary change. 

There turned out to be more code that had to be tweaked, because of the change to `metadata/default` as the source of truth. See [560b1dd](https://github.com/woocommerce/woocommerce-ios/pull/6936/commits/560b1dd7bb17c0d5a2484bd0c5e7b9fc3352a97c).

I verified it works correctly by running the lane on a demo branch. Worth noticing that the lane as it is doesn't commit the files it downloads or removes. Given the tweaks we have planned, I didn't spend extra time to amend that.

cc @rachelmcr

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -29,9 +28,46 @@ SCREENSHOT_DEVICES = [
+}.freeze
+
+# URL of the GlotPress project containing the strings used in the app
+GLOTPRESS_APP_STRINGS_URL = 'https://translate.wordpress.com/projects/woocommerce/woocommerce-ios/'
```
Worth remembering to unify those constant names with the new names you recently used in WPiOS for the similar ones (https://github.com/wordpress-mobile/WordPress-iOS/pull/18802). Up to you to do it in this PR or in a subsequent one, but just wanted to note it as a reminder.

mokagio replied:
Yep. 

https://github.com/woocommerce/woocommerce-ios/pull/7031

I actually finished that PR before seeing this comment 👉 🧠

AliSoftware replied:
👌🧠

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -785,33 +832,14 @@ platform :ios do
-
     # Copy source files into `en-US`
     enUS_path = File.join(Dir.pwd, 'metadata', 'en-US')
 
```
That comment is a bit misleading now that `metadata/default` is the new source of truth for ASC metadata, and made me pause (and then spiral into [a braindump below](https://github.com/woocommerce/woocommerce-ios/pull/6936#discussion_r886724474) 😅 )
```suggestion
    # Copy appstoreres (screenshot-related) txt files into `en-US`
    enUS_path = File.join(Dir.pwd, 'metadata', 'en-US')
```

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -829,11 +857,6 @@ platform :ios do
-    # Replace the `default` directory with the contents of `en-US`
-    default_path = File.join(Dir.pwd, 'metadata', 'default')
-    FileUtils.remove_dir(default_path) if File.directory?(default_path)
-    FileUtils.copy_entry(enUS_path, default_path)
```
> There turned out to be more code that had to be tweaked, because of the change to metadata/default as the source of truth. See [560b1dd](https://github.com/woocommerce/woocommerce-ios/pull/6936/commits/560b1dd7bb17c0d5a2484bd0c5e7b9fc3352a97c).

Took me a while to put together everything in my head to double-check this, so I'm writing this down for posterity below. The TL;DR is:
 - We should delete the `fastlane/metadata/default/promo_screenshots_*.txt` files as part of this PR
 - We might want to fix the inconsistency in the `whats-new` key between the 2 lanes

### Notes on `default` vs `en-US` for screenshot copies

 - 👍  We indeed made `fastlane/metadata/default` be the new source of truth for `release-notes.txt` via [those lines](https://github.com/woocommerce/woocommerce-ios/blob/trunk/fastlane/Fastfile#L721-L726) in a previous PR
 - 👍  `fastlane/metadata/default` is also the new source of truth for other App Store metadata files like `title.txt` and the like: those are not modified/copied/manipulated by our `Fastfile` as `release_notes.txt` is, but instead we're supposed to modify the `title.txt` files and the like manually, directly where they already are in `fastlane/metadata/default` (they are not present in `fastlane/metadata/en-US`, and we even check that they aren't via [those lines](https://github.com/woocommerce/woocommerce-ios/blob/trunk/fastlane/Fastfile#L746-L751))
 - ⚠️  On the other end, the source of truth for the `promo_screenshot_*.txt` files as still, for now at least, in `fastlane/appstoreres/metadata/source/` (we hope to fix this to put them in `fastlane/metadata/default` or `fastlane/metadata/en-US` (†) as well at some point, but that's for another and future PR)
    - But as of today, **both** `fastlane/metadata/default` AND `fastlane/metadata/en-US` contains `promo_screenshot_*.txt` files.
    - Those lines you deleted above are likely the reason for both of those to exist (because up until now those lines copied the files over last time the lane was run)
    - ⚠️  I'm completely fine with getting rid of those lines, especially given `default/` is not used by our screenshot compositor (†) but then **we should also delete the `fastlane/metadata/default/promo_screenshots_*.txt` files from git as well as part of this PR**, to only keep `fastlane/metadata/en-US/promo_screenshots_*.txt` (which will be updated/overwritten with the ones from `appstoreres/metadata/source` as part of the lines above next time we run this lane)

### Note: The Screenshots Compositor is not using `default`

(†) As a side note, given those `promo_screenshot_*.txt` files are used by [our screenshots compositor](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/docs/screenshot-compositor.md) (as opposed to being used by `fastlane`/`deliver` itself), I don't think our compositor does have the similar behavior of "falling back to `default` for translations that are missing for a given locale" like `deliver` does.

Which means that [when we'll get rid of that `download_app_store_translations ` lane in a later PR to only keep `download_localized_metadata_from_glotpress`](https://github.com/woocommerce/woocommerce-ios/blob/trunk/fastlane/Fastfile#L773-L775), we should be careful to copy ASC metadata files used by `deliver` (like `release_notes.txt` and `title.txt`) in `metadata/default`, BUT still keep screenshot files used by our compositor (like `promo_screenshot_*.txt`) in `en-US`, not `default` (especially if in that future, we also use the occasion to finally get rid of `fastlane/appstoreres/metadata/source/promo_screenshot_*.txt` and use `fastlane/metadata/en-US/promo_screenshot_*.txt` (and not `default/promo_screenshot_*.txt`!) as the direct source of truth)

_PS: I tried to make sense of [what we meant by those comments in the implementation of the `promo_screenshots` action](https://github.com/wordpress-mobile/release-toolkit/blob/d80471182e80d151773153536bcfc590e83a962c/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/promo_screenshots_action.rb#L28-L38) but they confused me more than anything, and I think that would not account for any fallback to `default` in case of missing localizations anyway._

### Note on `whats-new` key inconsistency

🐛  Note that this `download_app_store_translations` lane (which is focused on screenshots metadata) [uses `whats-new` as the GP key to extract from GlotPress](https://github.com/woocommerce/woocommerce-ios/blob/trunk/fastlane/Fastfile#L783), while the main lane `download_localized_metadata_from_glotpress` (the one we're likely to ultimately keep once we address [this FIXME](https://github.com/woocommerce/woocommerce-ios/blob/trunk/fastlane/Fastfile#L773-L775)) [uses `v<version>-whats-new`](https://github.com/woocommerce/woocommerce-ios/blob/trunk/fastlane/Fastfile#L731).

Even if we'll likely delete this `download_app_store_translations` soon (as per its FIXME) should we still fix this here, at least to have something consistent in the meantime (given we're not sure _when_ we'll get to address that FIXME so better have something coherent in the meantime)?

---

_(Sorry for the long comment, but it also helped me write down my reasoning here because I started to get lost in the various contexts of those different `.txt` files and their source of truth and the pending FIXMEs and all… so figured I might as well share my braindump to help and confirm my understanding)_

mokagio replied:
> Sorry for the long comment, but it also helped me write down my reasoning here because I started to get lost in the various contexts of those different .txt files and their source of truth and the pending FIXMEs and all…

There's nothing like writing things down to make sense of things. Or to make sense of how they do not make sense 🙃 Thank you for sharing. Super useful.

AliSoftware reviewed (commented):
I think we should delete the `fastlane/metadata/default/promo_screenshots_*.txt` files as part of this PR, as a follow up of your commit 560b1dd. See inline comment below for details.

We might also want to fix the inconsistency in the `whats-new` key between the 2 lanes, even if the 2nd lane will hopefully soon disappear when we get time to address its FIXME and unify the two.

mokagio committed:
Fix typo in a comment

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio committed:
Delete unused `fastlane/metadata/default/promo_screenshot_*.txt` files

`fastlane/metadata/default/` should be the source of truth for all
metadata, but we haven't yet updated the part of the logic that has to
do with the sceenshots localization and generation.

Until we do that, having those file in there is confusing and
wasteful, because they are never read as the source of truth for what
they stand for, which is instead in
`fastlane/appstoreres/metadata/source`.

This is a followup to the work in 560b1dd7bb17c0d5a2484bd0c5e7b9fc3352a97c

See discussion at
https://github.com/woocommerce/woocommerce-ios/pull/6936#discussion_r886724474

mokagio committed:
Use consistent key for "what's new" in `download_app_store_translations`

Consistent with the other lane doing a similar job,
`download_localized_metadata_from_glotpress`.

See discussion at
https://github.com/woocommerce/woocommerce-ios/pull/6936#discussion_r886724474

mokagio committed:
Clarify a screenshots localization comment

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio commented:
@AliSoftware, thank you for your detailed review. This should be ready for another look.

> I think we should delete the fastlane/metadata/default/promo_screenshots_*.txt files as part of this PR, as a follow up of your commit https://github.com/woocommerce/woocommerce-ios/commit/560b1dd7bb17c0d5a2484bd0c5e7b9fc3352a97c. See inline comment below for details.

[`aaba58c`](https://github.com/woocommerce/woocommerce-ios/pull/6936/commits/aaba58c071e10e8070586417cf2e8c4eae6e2b6a)

We might also want to fix the inconsistency in the whats-new key between the 2 lanes, even if the 2nd lane will hopefully soon disappear when we get time to address its FIXME and unify the two.

> [`7feedea`](https://github.com/woocommerce/woocommerce-ios/pull/6936/commits/7feedea2ea2fe117a8d35a398623cafc06db94a4)

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -911,8 +934,38 @@ platform :ios do
   end
 
   desc 'Run localization only'
```
Nit: Worth making this `desc` a bit more descriptive? 🤷 

```suggestion
  desc 'Updates the main `Localizable.strings` file — that will be imported by GlotPress — from code and `Info.plist` files'
```

mokagio replied:
Might as well while we're at it 👍

AliSoftware reviewed (approved):
Just left a suggestion for better `desc` for the lane, otherwise LGTM 👍 🎉

mokagio committed:
Improve `generate_strings_file_for_glotpress` `desc`

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>
